src/redux/walletconnect.ts

```diff
@@ -474,11 +474,7 @@ const listenOnNewMessages =
         const { chainId } = payload.params[0];
         // @ts-expect-error "_chainId" is private.
         const currentChainId = Number(walletConnector._chainId);
-        const supportedChains = RainbowNetworkObjects.filter(network => network.features.walletconnect).map(network =>
-          network.id.toString()
-        );
-        const numericChainId = convertHexToString(chainId);
-        if (supportedChains.includes(numericChainId)) {
+        if (supportedWalletConnectChainIds.includes(currentChainId)) {
```

Comment:
`currentChainid` can be different from the `numericChainId` that we get from the payload - the original intent was to check the `chainId` of the message payload so this changes behavior

Reply:
also remember that you'll need to still convert from hex: `convertHexToString(chainId);` when parsing the payload

Reply:
ok rolling back to use the one from payload